### DISCUSSION TITLE
Weird thing with satin stitch

### DISCUSSION THREAD

**Discussion by dragonlord66666** [inkstitch/inkstitch#1116] [Q&A]
I asked Lex this question
 I have the outline of a jelly baby as a stroke 2mm wide and when I go line to stain I get two paths that are the outline cut in half, not one single outline as a satin stitch?
could this be a setting changed? is there an easy way to reset back to defaults?

Lex answered me with 
The Convert To Satin might break up the line into multiple satins because of how the code works.  If your line crosses over itself or if it ends in the same spot that it starts, that will cause Ink/Stitch to give you two or more satins as a result.  

They are arranged one after the other, so they should sew out one right after the other.  If you really want, you can combine the resulting objects, join the rails, and then you'd have just one satin.

____

thing was it was stitching the same side (left) about three times in satin stitch it should have done one 2mm wide satin. 

I reinstalled inkstitch (I have Mojave on my mac) finally - it never puts the inkstitch XXX.py or inx files in the right place, it places them inside a folder called inkstitch-v1.26.2-osx-mojave-x86_64-en_US with a inkstitch folder and I have to drag them into place but...- 
I started over with the design, I'm doing Appliqué so put the outline, I named it placement and made it a dashed line, then duplicated the object and called that tackdown then duplicated it again and made that a solid line 3mm wide and now I'm getting massive satin stitch (see pic) I paused it of course
![Screenshot 2021-04-02 at 9 41 26 pm](https://user-images.githubusercontent.com/81831005/113452805-b1dca080-93fc-11eb-9af0-9dc1b9abfdfa.png)
what would cause this? as that is NOT 3mm satin stitch LOL
thanks

thought I'd add a pic of the inkscape screen.
![Screenshot 2021-04-02 at 10 50 03 pm](https://user-images.githubusercontent.com/81831005/113456837-3a137380-9406-11eb-8410-7a8229a1f509.png)

**Comment by wwderw** [comment]
My suggestion (and unfortunately this would take longer to do) would be to digitize it manually as a satin stitch, but don't use just a single stroke, but digitizing the edges of the satin stitch, add in your stitch angles, combine all the objects together and then apply your params that way.

It does add more time to the digitizing process, but it helps the program figure out where you are going with everything and leaves less for the program to figure out on it's own.

Also considering this is for applique, I would actually suggest an 8mm cover stitch, if that is done, then those corners would be better digitizing as separate objects versus all one single object as it is.  The reason for the extra wide cover stitch is to make sure that the applique fabric is grabbed and stitched over and not having the edges peak out etc.

**Reply by dragonlord66666** [reply]
tried 8mm and get this.
![Screenshot 2021-04-02 at 11 00 34 pm](https://user-images.githubusercontent.com/81831005/113457280-5663e000-9407-11eb-8454-4559adb36f7c.png)
I'm not understanding why it's doing it now after a new install, it worked before (apart from the splitting a shape into two but) plus that is no way 8mm statin stitch
only different seems to be how far above the shape it starts the satin - which would be right if it only went as a smaller zig zag.

**Reply by wwderw** [reply]
Oh no, you'll have to change how you digitize the cover stitch.  Don't use then center line stroke method, but digitize where and how you want the satin stitch boundaries and apply stitch angles accordingly.  Merge all the objects and then apply your params.

**Reply by dragonlord66666** [reply]
I don't use centre line trace. I used bezier tool to create the shape. 
thing is, it worked before so why is it now NOT working. I really should not have to manually set satin stitches by, if I understand you right, making a double outline and setting them the width apart I want.
just thought, only thing different is the location it's installed in.
originally I had it in /Applications/Inkscape 1-0.app/Contents/Resources/share/inkscape/extensions
but the page says to put it into the user extensions folder which I did

**Reply by wwderw** [reply]
I wasn't talking about trace, I was just talking about using a stroke to generate the satin stitch width.  It was just a suggestion to get back up and running quickly as one can while waiting for a fix or at least the cause of issue.

Location of the binary shouldn't make a difference, but I don't really change location of where it's installed so I can't say for sure.  Just so long as you put it somewhere where Inkscape knows to look for extensions.  

Are you using the exact same Ink/Stitch binaries or did you swap it with a different set when doing the reinstall?

**Reply by dragonlord66666** [reply]
that might be it. was just looking and I saved the install file I use... the one from inkstitch for mojave is inkstitch-v1.26.2-osx-mojave-x86_64-en_US.tar.gz the original I just found is inkstitch-refs-tags-v1.28.0-osx-en_US.zip so I'll uninstall (remove) the inkstitch I just put on (1.26.2), and reinstall the old one (1.28.0) I had.
see what happens.

**Reply by dragonlord66666** [reply]
well I'm back to regular satin stitch using the inkstitch?convert line to satin - YEAH!!
![Screenshot 2021-04-03 at 1 59 37 am](https://user-images.githubusercontent.com/81831005/113463848-4dcbd380-9420-11eb-9eaa-e5bd3bbb2158.png)

so back to original problem where it splits the shape into two - but so long as it's stitching  as one object, it is, I can live with that - so when it was stitching half the satin design three times it either had a brain fart or a setting changed I am unaware of.
![Screenshot 2021-04-03 at 1 56 09 am](https://user-images.githubusercontent.com/81831005/113463868-7784fa80-9420-11eb-9447-58dd23934a98.png)

think I might save current params as default. so then I should, in theory, be able to reset easier than a delete and reinstall (although I guess I could just overwrite the files from the zip file)

**Reply by dragonlord66666** [reply]
when I try to join the two halves I join the nodes on the inside track and then the outside track but I then get errors

Satin2 (path2): error: Satin column: A rung doesn't intersect both rails. Each rung should intersect both rails once.
Traceback (most recent call last):
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/SocketServer.py", line 599, in process_request_thread
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/SocketServer.py", line 334, in finish_request
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/SocketServer.py", line 655, in __init__
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/werkzeug/serving.py", line 329, in handle
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/BaseHTTPServer.py", line 340, in handle
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/werkzeug/serving.py", line 364, in handle_one_request
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/werkzeug/serving.py", line 306, in run_wsgi
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/werkzeug/serving.py", line 294, in execute
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/flask/app.py", line 1997, in __call__
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
  File "/Users/runner/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
  File "/Users/runner/runners/2.169.1/work/inkstitch/inkstitch/lib/api/stitch_plan.py", line 14, in get_stitch_plan
  File "/Users/runner/runners/2.169.1/work/inkstitch/inkstitch/lib/extensions/base.py", line 185, in elements_to_patches
  File "/Users/runner/runners/2.169.1/work/inkstitch/inkstitch/lib/elements/element.py", line 307, in embroider
  File "/Users/runner/runners/2.169.1/work/inkstitch/inkstitch/lib/elements/element.py", line 366, in validate
  File "/Users/runner/runners/2.169.1/work/inkstitch/inkstitch/lib/elements/element.py", line 333, in fatal
SystemExit: 1

so I guess joining the nodes is not the right thing to do....or I need to do something else to fix the 'rung' error
would be nice to have the outline satin as a single object but, meh...

**Comment by lexelby** [comment]
Ah!  That picture is what happens when one of the rails is reversed.  It probably happened when you tried to join the satins together.

There's an open in Inkscape's settings for the node editor that will show you the direction of paths.  Enable that, then use the node editor tool and examine the satin.  Select a point in one of the rails and press ctrl-r to reverse it.

**Reply by dragonlord66666** [reply]
I will have a look at that. I've saved as an embroidery file as it does seem to sew out ok - just need to wait for my machine to arrive....
might have a play with start and stop command once I get it all set up. think it would be nice if it started and stopped at the top of the head. at moment the placement & tackdown start on the left leg/thigh area and the satin starts in the middle of the legs LOL

**Reply by lexelby** [reply]
Oh, that could be another problem.  With satin stitch, you tell Ink/Stitch where to start and stop by having the rails be broken in one spot.  If you join _both_ ends to make a full loop, Ink/Stitch won't know where you meant it to start.

Technically speaking, Inkscape decides in that case, and it's almost always not what you want.  One rail may start in one place and the other may start in an entirely different place, which will make things really weird.

**Reply by dragonlord66666** [reply]
so should I have made the jelly baby shape but broken apart the nodes at the top of the head (or where ever I wanted stitching to start)?

**Reply by dragonlord66666** [reply]
well splitting the design at the top node works, for running stitches it's fine but for the satin stitch it's back to the massive stitches. think inkstitch just likes to make my life harder LOL

is there a reason I can't duplicate the tackdown shape/layer, change the stroke to a solid line (and change the colour for ease viewing) and then convert line to satin - am I missing something?

![image](https://user-images.githubusercontent.com/81831005/113487350-85ce2780-94af-11eb-83e6-1920c1013f61.png)

**Reply by dragonlord66666** [reply]
OK no idea what I did but kept deleteing the satin layer an re doing it. even got it stitching out the individual rails and the tracks..... but I finally go it to work. (long may that continue)

![image](https://user-images.githubusercontent.com/81831005/113489087-f7ab6e80-94b9-11eb-9f7b-4ed2d4ca5c92.png)


now I just need to add details inside and hopefully I can save as final embroidery.

**Comment by tyrosinase** [comment]
On 4/3/21 3:24 PM, dragonlord66666 wrote:
>
> OK no idea what I did but kept deleteing the satin layer an re doing 
> it. even got it stitching out the individual rails and the tracks..... 
> but I finally go it to work. (long may that continue)
>

Yep, the long-stitches thing (I call it "string art mode") is just when 
one rail goes the wrong direction. Click one rail with the node edit 
tool so you're just selecting a segment of it, then pick Reverse Path 
(so it reverses only that subpath, not the entire shape) and it will 
stitch correctly.

**Reply by dragonlord66666** [reply]
ah, not sure how it kept swapping direction but now I know a trick to get it back. thanks

next up. learn to digitise fur for an animal patch...... so fur AND a colour image.

**Comment by tyrosinase** [comment]
On 4/3/21 9:07 PM, dragonlord66666 wrote:
>
> ah, not sure how it kept swapping direction but now I know a trick to 
> get it back. thanks
>

Inkscape (not Ink/Stitch) will reverse the [sub]path if you so much as 
look at it funny. For instance: Use the bezier tool to click on the end 
of your path and draw an extension to it? That is now the direction that 
path goes. It makes sense but it takes a little getting used to.

I forget if it's the default but if not, go into `Edit > Preferences > 
Tools > Node` and check `Show path direction on outlines.` That'll give 
you little arrows on the outlines when you're in node-editing mode. Very 
useful for seeing which direction the stitching is going to go, and for 
making sure your satin rails are going the same direction.
